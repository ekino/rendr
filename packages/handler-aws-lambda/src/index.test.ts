import { readFileSync } from "fs";
import { createContext } from "./index";

describe("create context", () => {
  function loadJson(path: string) {
    return JSON.parse(readFileSync(path, { encoding: "utf8" }));
  }

  it("createContext from GET request", async () => {
    const payload = loadJson(`${__dirname}/__fixtures__/payload_get_v2.json`);

    const ctx = createContext(payload.event, {});

    expect(ctx).toMatchSnapshot();
  });

  it("createContext from POST request", async () => {
    const payload = loadJson(`${__dirname}/__fixtures__/payload_post_v2.json`);

    const ctx = createContext(payload.event, {});

    expect(ctx).toMatchSnapshot();
  });
});
