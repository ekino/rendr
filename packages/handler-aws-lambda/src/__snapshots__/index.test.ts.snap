// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create context createContext from GET request 1`] = `
Object {
  "isClientSide": false,
  "isServerSide": true,
  "req": Object {
    "body": undefined,
    "headers": Object {
      "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
      "accept-encoding": "gzip",
      "accept-language": "en,en-US;q=0.8,fr;q=0.5,fr-FR;q=0.3",
      "cookie": "__cfduid=XXXXXXXXXX",
      "dnt": "1",
      "early-data": "1",
      "host": "example.com",
      "upgrade-insecure-requests": "1",
      "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:81.0) Gecko/20100101 Firefox/81.0",
      "x-amzn-trace-id": "Root=Z-XXXXXXXXX-XXXXXXXXXXXXXXX",
      "x-forwarded-for": "127.0.0.1",
      "x-forwarded-port": "443",
      "x-forwarded-proto": "https",
    },
    "hostname": "example.com",
    "method": "GET",
    "params": Object {},
    "pathname": "/info",
    "query": Object {},
  },
  "settings": Object {},
}
`;

exports[`create context createContext from GET request with empty path 1`] = `
Object {
  "isClientSide": false,
  "isServerSide": true,
  "req": Object {
    "body": undefined,
    "headers": Object {
      "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
      "accept-encoding": "gzip",
      "accept-language": "en,en-US;q=0.8,fr;q=0.5,fr-FR;q=0.3",
      "cookie": "__cfduid=XXXXXXXXXX",
      "dnt": "1",
      "early-data": "1",
      "host": "example.com",
      "upgrade-insecure-requests": "1",
      "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:81.0) Gecko/20100101 Firefox/81.0",
      "x-amzn-trace-id": "Root=Z-XXXXXXXXX-XXXXXXXXXXXXXXX",
      "x-forwarded-for": "127.0.0.1",
      "x-forwarded-port": "443",
      "x-forwarded-proto": "https",
    },
    "hostname": "example.com",
    "method": "GET",
    "params": Object {},
    "pathname": "/",
    "query": Object {},
  },
  "settings": Object {},
}
`;

exports[`create context createContext from POST request 1`] = `
Object {
  "isClientSide": false,
  "isServerSide": true,
  "req": Object {
    "body": Object {
      "foo": "bar",
    },
    "headers": Object {
      "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
      "accept-encoding": "gzip",
      "accept-language": "en,en-US;q=0.8,fr;q=0.5,fr-FR;q=0.3",
      "content-type": "application/x-www-form-urlencoded",
      "cookie": "__cfduid=XXXXXXXXXX",
      "dnt": "1",
      "early-data": "1",
      "host": "example.com",
      "upgrade-insecure-requests": "1",
      "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:81.0) Gecko/20100101 Firefox/81.0",
      "x-amzn-trace-id": "Root=Z-XXXXXXXXX-XXXXXXXXXXXXXXX",
      "x-forwarded-for": "127.0.0.1",
      "x-forwarded-port": "443",
      "x-forwarded-proto": "https",
    },
    "hostname": "example.com",
    "method": "POST",
    "params": Object {},
    "pathname": "/info",
    "query": Object {},
  },
  "settings": Object {},
}
`;
